streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -690,7 +691,7 @@ private int restoreChangelog(final Task task, final ChangelogMetadata changelogM
             }
         }
 
-        if (numRecords > 0 || changelogMetadata.state().equals(ChangelogState.COMPLETED)) {
+        if (task != null && (numRecords > 0 || changelogMetadata.state().equals(ChangelogState.COMPLETED))) {
```

Comment:
In https://github.com/apache/kafka/pull/13300/files#r1123945673 , we discussed if the null check is necessary, and then decided to drop it. But it turns out it's necessary. Otherwise, NPE will be thrown in some test cases.